src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/IdentifierOfTypeSearchValue.cs

```diff
@@ -0,0 +1,109 @@
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IdentifierOfType"/> class.
+        /// </summary>
+        /// <param name="system">The identifier system value.</param>
+        /// <param name="code">The identifier code value.</param>
+        /// <param name="value">The identifier value.</param>
+        public IdentifierOfTypeSearchValue(string system, string code, string value)
+        {
+            // All values must be present and non empty.
+            EnsureArg.IsTrue(
+                !string.IsNullOrWhiteSpace(system) &&
+                !string.IsNullOrWhiteSpace(code) &&
+                !string.IsNullOrWhiteSpace(value));
+
+            System = system;
+            Code = code;
+            Value = value;
+        }
+
+        /// <summary>
+        /// Gets the identifier system value.
+        /// </summary>
+        public string System { get; }
+
+        /// <summary>
+        /// Gets the identifier code value.
+        /// </summary>
+        public string Code { get; }
+
+        /// <summary>
+        /// Gets the identifier value.
+        /// </summary>
+        public string Value { get; }
+
+        /// <inheritdoc />
+        public bool IsValidAsCompositeComponent => false;
+
+        /// <summary>
+        /// Parses the string value to an instance of <see cref="IdentifierOfTypeSearchValue"/>.
+        /// </summary>
+        /// <param name="s">The string to be parsed.</param>
+        /// <returns>An instance of <see cref="IdentifierOfTypeSearchValue"/>.</returns>
+        public static IdentifierOfTypeSearchValue Parse(string s)
+        {
+            EnsureArg.IsNotNullOrWhiteSpace(s, nameof(s));
+
+            IReadOnlyList<string> parts = s.SplitByTokenSeparator();
+
+            if (parts.Count == 3)
+            {
+                return new IdentifierOfTypeSearchValue(parts[0], parts[1], parts[2]);
+            }
+            else
+            {
+                throw new FormatException(Resources.IdentifierOfTypeMustHaveThreeParams);
+            }
+        }
+
+        /// <inheritdoc />
+        public void AcceptVisitor(ISearchValueVisitor visitor)
+        {
+            EnsureArg.IsNotNull(visitor, nameof(visitor));
+
+            visitor.Visit(this);
+        }
+
+        public bool Equals([AllowNull] ISearchValue other)
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            var identifierSearchValueOther = other as IdentifierOfTypeSearchValue;
+
+            if (identifierSearchValueOther == null)
+            {
+                return false;
+            }
+
+            return System.Equals(identifierSearchValueOther.System, StringComparison.OrdinalIgnoreCase) &&
+                   Code.Equals(identifierSearchValueOther.Code, StringComparison.OrdinalIgnoreCase) &&
+                   Value.Equals(identifierSearchValueOther.Value, StringComparison.OrdinalIgnoreCase);
```

Comment:
Should this be case sensitive since Token is case sensitive? https://www.hl7.org/fhir/search.html#token

Reply:
I copied it from TokenSearchValue in hope it's coded right.

Who would though it didn't :)

Reply:
I actually have no idea what it for, and why it's in all SearchValues.